### PR TITLE
Yuhsuan/2d fitting

### PR DESCRIPTION
**Pull request by YuHsuan-Hwang**
Add messages for Gaussian image fitting. Companion PR for https://github.com/CARTAvis/carta-backend/pull/1055.

### CODE REVIEW

File: shared/defs.proto [hunk 1]
```diff
@@ -194,4 +194,32 @@ message ListProgress {
...
     sfixed32 checked_count = 3;
     sfixed32 total_count = 4;
-}
\ No newline at end of file
+}
+
+// parameters of a 2D Gaussian component for image fitting
+message GaussianComponent {
+    // x coordinate of the center in pixels
+    double center_x = 1;
```
Comment (veggiesaurus) [inline shared/defs.proto]: We have a sub-message `Point` for x,y pairs. I think we should use that, unless they specifically need to be double-precision? Even so, we could make a new message `DoublePoint`

File: shared/defs.proto [hunk 2]
```diff
@@ -194,4 +194,32 @@ message ListProgress {
...
+    // y coordinate of the center in pixels
+    double center_y = 2;
+    // amplitude of the component
+    double amp = 3;
+    // full width at half maximum along x coordinate in pixels
+    double fwhm_x = 4;
```
Comment (veggiesaurus) [inline shared/defs.proto]: same comment about using `Point`

**Review by veggiesaurus — CHANGES_REQUESTED** [review]
One change requested. I'd also like to see some information added to the documentation, perhaps in the form of a sequence diagram? We've been very lax about this. Reference example: https://carta-protobuf.readthedocs.io/en/latest/region_selection_and_statistics.html#per-cube-histograms

File: shared/defs.proto (hunk 1 above)
Comment (YuHsuan-Hwang) [reply shared/defs.proto]: I think float would be sufficient. will change to `Point`

**Comment by YuHsuan-Hwang** [comment]
> One change requested. I'd also like to see some information added to the documentation, perhaps in the form of a sequence diagram? We've been very lax about this. Reference example: https://carta-protobuf.readthedocs.io/en/latest/region_selection_and_statistics.html#per-cube-histograms

The current procedure of image fitting is relatively simple, but I think it's good to add more information in the document. will add a sequence diagram.

**Comment by YuHsuan-Hwang** [comment]
@veggiesaurus I added a page in the behaviour section. The content may be extended if we add more features in the future.

**Pull request merged**